### PR TITLE
Update the README and Add Error Checking Code

### PR DESCRIPTION
**Pull request by jxguan**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -73,19 +73,31 @@ func (c *Client) AddFile(filename string) error {
 	if _, found := c.reverseLookup[file]; found {
 		return errors.New("file already exists")
 	}
-	content, _ := ioutil.ReadFile(filename)
-	docID := c.server.AddFile(content)
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	docID, errAddFile := c.server.AddFile(content)
```
Comment (strib) [inline client/client.go]: You can reuse the `err` variable everywhere, no reason not to.

File: client/client.go [hunk 2]
```diff
@@ -113,7 +125,7 @@ func (c *Client) getFile(docID int) {
 	if _, err := os.Stat(filename); err == nil {
 		return
 	}
-	content := c.server.GetFile(docID)
+	content, _ := c.server.GetFile(docID)
```
Comment (strib) [inline client/client.go]: Why not check this error?

File: client/client_test.go [hunk 3]
```diff
@@ -112,7 +112,7 @@ func TestAddFile(t *testing.T) {
 		t.Fatalf("lookup table not set up correctly on the server")
 	}
 
-	if !bytes.Equal(s.GetFile(0), []byte(content)) {
+	if actual, err := s.GetFile(0); err != nil || !bytes.Equal(actual, []byte(content)) {
```
Comment (strib) [inline client/client_test.go:115]: Good practice to add `err` to the fatal message.

File: server/server.go [hunk 4]
```diff
@@ -154,7 +160,11 @@ func (s *Server) WriteIndex(si index.SecureIndex) error {
 
 // readIndex loads an index from the disk.
 func (s *Server) readIndex(docID int) (si index.SecureIndex, err error) {
-	input, _ := ioutil.ReadFile(path.Join(s.directory, strconv.Itoa(docID)+".index"))
+	input, errInput := ioutil.ReadFile(path.Join(s.directory, strconv.Itoa(docID)+".index"))
```
Comment (strib) [inline server/server.go]: You can just call it `err` and do a `return err` below.

File: server/server_test.go [hunk 5]
```diff
@@ -95,13 +95,16 @@ func TestAddAndGetFile(t *testing.T) {
 		[]byte("Now the third.")}
 
 	for i := 0; i < 3; i++ {
-		if s.AddFile(files[i]) != i {
-			t.Fatalf("incorrect number of files returned")
+		if docID, err := s.AddFile(files[i]); docID != i || err != nil {
+			t.Fatalf("error when adding files")
```
Comment (strib) [inline server/server_test.go]: Print `err`.

File: server/server_test.go [hunk 6]
```diff
@@ -95,13 +95,16 @@ func TestAddAndGetFile(t *testing.T) {
...
 		}
 	}
 
 	for i := 0; i < 3; i++ {
-		content := s.GetFile(i)
+		content, err := s.GetFile(i)
+		if err != nil {
+			t.Fatalf("error when getting the files")
```
Comment (strib) [inline server/server_test.go]: Print `err`.

**Comment by strib** [comment]
Looks good to me mod nits, thanks!

File: client/client.go (hunk 1 above)
Comment (jxguan) [inline client/client.go]: Fixed

Ah I thought for := every variable on the left need to be undeclared..

File: client/client_test.go (hunk 3 above)
Comment (jxguan) [inline client/client_test.go:115]: Fixed

File: server/server_test.go (hunk 5 above)
Comment (jxguan) [inline server/server_test.go]: Fixed

File: server/server_test.go (hunk 6 above)
Comment (jxguan) [inline server/server_test.go]: Fixed

File: server/server.go (hunk 4 above)
Comment (jxguan) [inline server/server.go]: Fixed

**Pull request merged**